main/utils/trophy.h

```diff
@@ -0,0 +1,451 @@
+disabled unless explicitly set active by a developer. To activate it, create the following structs in the .c file for
+the mode:
+ * \code {.c}
+const trophyData_t exampleTrophies[] = {
+    {
+        .title       = "Trigger Trophy",
+        .description = "You pressed A!",
+        .imageString = "kid0.wsg",
+        .type        = TROPHY_TYPE_TRIGGER,
+        .difficulty  = TROPHY_DIFF_EASY,
+        .maxVal      = 1, // For trigger type, set to one
+    },
+}
+
+trophySettings_t exampleTrophySettings = {
+    .drawFromBottom   = false,
+    .staticDurationUs = DRAW_STATIC_US,
+    .slideDurationUs  = DRAW_SLIDE_US,
+};
+ trophyDataList_t trophyTestData = {.settings = &trophyTestModeTrophySettings,
+                                    .list = trophyTestModeTrophies,
+                                    .length = ARRAY_SIZE(trophyTestModeTrophies)};
+
+swadgeMode_t trophyTestMode = {.modeName      = modeName,
+                               .wifiMode      = NO_WIFI,
+                                //...
+                               .fnAdvancedUSB = NULL,
+                               .trophyData    = &trophyTestData};
+ * \endcode
+ *
+ * This will allow the trophy system to access your trophies and make sure the right text and images are displayed when
+a trophy is triggered. Also, add the setting data as `extern` to ensure that the data can be accessed by the trophy case
+and swadgepass
+ *
+ * \section trophy_settings Settings
+ *
+ * There are four settings that can be set for the mode overall:
+ * - drawFromBottom: If set to true, draws the banner at the bottom
+ * - staticDurationUs: How long the banner stays on screen for. The default value (DRAW_STATIC_US) is about half a
+second
+ * - slideDurationUS: How long it takes for the banner to slide in and out. Set to (DRAW_SLIDE_US) for quarter second
+slide, or 0 to have the banner pop in.
+ * - namespaceKey: A text string used to save data to the NVS. This is set automatically if left blank, and should only
+be manually set when there's a conflict between modes
+ *
+ * These can be loaded into the settings struct as shown above.
+ *
+ * \section trophy_individual Individual trophy settings
+ *
+ * The largest portion of the work required to set up the trophies it to create all the trophies from scratch. Here's
+the types of data available to set for each individual trophy:
+ * - Title: The name of the trophy. Make it unique! This is used as the identifier and having two of the same names will
+ * confuse the program. There's no need to worry about sharing the name with other modes though, that's handled
+ * internally.
+ * - Description: Displays text to help the player figure out how to get the trophy. Text-box size is limited, and too
+ * long descriptions will get truncated. Descriptions can be off-topic, but may confuse players.
+ * - Image string: Provide a 36x36 sprite that will be displayed. If string is empty (""), the system will not draw and
```

Comment:
I mentioned it before, but it's probably better to have four default trophy icons for the four `trophyDifficulty_t`. Text-only feels like a generic notification, not a trophy thingy.

Reply:
Sure, I'm all for a "no image" option, I just don't think that should be the default.